docs/book/developer-guide/repo-and-config.md

```diff
@@ -2,8 +2,309 @@
+    user_metadata: null
+    version: 0.7.3
+    ```
+
+2. The `local_stores` directory is where some "local" flavors of Stack Components,
+such as the `local` Artifact Store, the `sqlite` Metadata Store or the `local`
+Secrets Manager persist data locally. Every local Stack Component will have its
+own subdirectory here named after the Stack Component's unique UUID. One notable
+example is the `local` Artifact Store flavor, that, when part of the active Stack,
+stores all the artifacts generated by Pipeline runs in the designated local
+directory.
+
+3. The `profiles` directory is used as a default root path location where ZenML
+stores information about the Stacks, Stack Components, custom Stack Component
+Flavors etc. that are configured under each Profile. Every Profile will have its
+own subdirectory here, unless the Profile is explicitly created with a custom
+configuration path (see the `zenml profile` command and the section on
+[ZenML Profiles](./profiles.md) for more information about Profiles).
+
+In addition to the above, you may also find the following files and folders under
+the _Global Config Directory_, depending on what you do with ZenML:
+
+* `zenml_examples` - used as a local cache by the `zenml example` command, where
+the pulled ZenML examples are stored.
+* `kubeflow` - this is where the Kubeflow orchestrators that are part of a Stack
+store some of their configuration and logs
+
+## The ZenML Repository
+
+A ZenML _Repository_ designates a folder that contains a complete set of Python
+scripts and modules as well as any associated files that are needed to run ZenML
+pipelines. Any such folder can be registered as a ZenML Repository root by running
+the `zenml init` command from that folder or passing its location as an argument
+to that same command, e.g.:
+
+```shell
+stefan@aspyre2:/tmp/zenml$ zenml init
+ZenML repository initialized at /tmp/zenml.
+The local active profile was initialized to 'default' and the local active stack to 'default'. This local configuration will only take effect when you're running ZenML from the initialized repository root, or from a subdirectory. For more information on profile and stack configuration, please visit https://docs.zenml.io.
+```
+
+The _Repository_ designation plays a double role in ZenML:
+
+* it is used by ZenML to identify which files it must copy into the environments
+it builds that are used to execute pipeline steps remotely, such as container
+images.
+* it stores local configuration parameters that can be set independently of the
+global configuration: the _Repository_ active profile and active stack. These
+settings are in effect when ZenML code is executed while the current working
+directory the _Repository_ root or one of its sub-folders. For more information
+on setting the active profile and stack local to a _Repository_, please visit
+the [ZenML Profiles](./profiles.md) section.
+
+A ZenML _Repository_ is easily identifiable by the presence of a `.zenml`
```

Comment:
```suggestion
A ZenML _Repository_ is easily identifiable by the presence of a (hidden) `.zen`
```

Not sure if it needs to be mentioned that it's hidden, but I guess in case someone goes there looking for it perhaps it's worth it.